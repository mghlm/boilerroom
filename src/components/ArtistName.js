import React from 'react';
import { Text, View, Image, TouchableOpacity } from 'react-native';
import ArtistBox from './ArtistBox';
import { Actions } from 'react-native-router-flux';

const ArtistName = ( {set} ) => {

  return (
    <TouchableOpacity onPress={ () => Actions.setDetails( {set: set} ) }>
      <ArtistBox>
        <View>
          <Text style={styles.titleTextStyle}>
            {set.artist.toUpperCase()}
          </Text>
        </View>
      </ArtistBox>
    </TouchableOpacity>

  );
};

const styles = {
  titleTextStyle: {
    fontSize: 30,
    color: "#FFF",
    fontFamily: 'Helvetica'
  }
}

export default ArtistName;
