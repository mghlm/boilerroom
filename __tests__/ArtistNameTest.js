import 'react-native'
import React from 'react';
import ArtistName from '../src/components/ArtistName';
// import { TouchableOpacity } from 'react-native'
// import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';


const testSet = {"artist": "test"};

test('renders correctly', () => {
  const tree = renderer.create(<ArtistName set={testSet}/>).toJSON();

expect(tree).toMatchSnapshot();
});
