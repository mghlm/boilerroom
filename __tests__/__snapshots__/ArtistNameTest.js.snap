// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "borderColor": "#7d7d7d",
        "borderWidth": 1,
        "justifyContent": "center",
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 10,
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontSize": 30,
          }
        }
      >
        TEST
      </Text>
    </View>
  </View>
</View>
`;
